File: python/data-wrangling-components/data_wrangling_components/engine/verbs/ungroup.py
#
# Copyright (c) Microsoft. All rights reserved.
# Licensed under the MIT license. See LICENSE file in the project.
#

from data_wrangling_components.table_store import TableContainer


def ungroup(input: TableContainer):
    """Ungroups a table.

    In pandas it converts a DataFrameGroupBy into a DataFrame object.

    :param step:
        Parameters to execute the operation.
        See :py:class:`~data_wrangling_components.engine.verbs.spread.SpreadArgs`.
    :type step: Step
    :param store:
        Table store that contains the inputs to be used in the execution.
    :type store: TableStore

    :return: new table with the result of the operation.
    """
    input_table = input.table
    output = input_table.obj
    return TableContainer(table=output)

File: javascript/workflow/src/verbs/ungroup.ts
/*!
 * Copyright (c) Microsoft. All rights reserved.
 * Licensed under the MIT license. See LICENSE file in the project.
 */
import type { ColumnTableStep } from './util/factories.js'
import { stepVerbFactory } from './util/factories.js'

export const ungroupStep: ColumnTableStep<void> = (input) => input.ungroup()
export const ungroup = stepVerbFactory(ungroupStep)
